### PR TITLE
fix(useActivator): Consider Focused status in isTop watcher

### LINKED ISSUE #15475: [Bug Report][3.0.0-beta.5] `v-tooltip` does not work with `open-on-focus` prop.
### Environment
**Vuetify Version:** 3.0.0-beta.5
**Vue Version:** 3.2.36
**Browsers:** Chrome 103.0.5060.114
**OS:** Windows 10

### Steps to reproduce
Try to put an `open-on-focus` prop.

### Expected Behavior
It should be open to focus 

### Actual Behavior
It not get opened.

### Reproduction Link
<a href="https://codepen.io/Alokdev/pen/PoRmqvL?editors=101" target="_blank">https://codepen.io/Alokdev/pen/PoRmqvL?editors=101</a>

### Other comments
It work when you put ` :open-on-hover="false"` prop with it.

### PR DESCRIPTION
**Pull request by yuwu9145**
## Description

fixes #15475

## Motivation and Context

fixes #15475

## How Has This Been Tested?

visually

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
<template>
  <v-app>
    <v-container>
      <div class="d-flex justify-space-around">
      <v-btn>
        click me and press tab
        <v-tooltip
          activator="parent"
          location="start"
        >Tooltip</v-tooltip>
      </v-btn>
  
      <v-btn>
        default
        <v-tooltip
          activator="parent"
          location="end"
        >Tooltip</v-tooltip>
      </v-btn>
  
      <v-btn>
        open on focus
        <v-tooltip
          activator="parent"
          location="top"
          open-on-focus
        >Tooltip</v-tooltip>
      </v-btn>
  
      <v-btn>
        default
        <v-tooltip
          activator="parent"
          location="bottom"
        >Tooltip</v-tooltip>
      </v-btn>
    </div>
      <!-- -->
    </v-container>
  </v-app>
</template>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by yuwu9145** [comment]
I will check out failed tests

File: packages/vuetify/src/components/VTooltip/__tests__/VTooltip.spec.cy.tsx [hunk 1]
```diff
@@ -11,4 +12,28 @@ const stories = {
...
+      </VBtn>
+    ))
+
+    cy.get('.v-btn').focus()
+
+    // This arbitrary time periods just aims to verify when UI is in stable state.
+    // As there seems to be no way for cypress to wait for an explicit
+    // vuetify/vue internal event finish
+    // eslint-disable-next-line
+    cy.wait(500)
```
Comment (yuwu9145) [inline packages/vuetify/src/components/VTooltip/__tests__/VTooltip.spec.cy.tsx:35]: This "arbitrary time periods" is an anti-pattern in cypress ([link](https://docs.cypress.io/api/commands/wait#Examples)).

I am stuck on finding a good way to allow cypress to wait for an explicit vuetify event (`isTop` watcher logic finish)

Please advice and thanks in advance

**Comment by KaelWD** [comment]
- Don't use props.openOnFocus directly, it is inferred based on other props by default
- This won't close nested menus properly unless they have both open-on-hover and open-on-focus

I came up with this but it keeps tooltips open if you hover then press tab:
```js
if (val && (
  (props.openOnHover && !isHovered && (!openOnFocus.value || !isFocused)) ||
  (openOnFocus.value && !isFocused && (!props.openOnHover || !isHovered))
)) {
```

**Comment by yuwu9145** [comment]
Thanks! I will check it out when got time

**Comment by KaelWD** [comment]
I think I'll just have to handle the hierarchy in a different way.

**Pull request closed without merging**